dashboard/app/models/code_review_comment.rb

```diff
@@ -32,10 +32,20 @@ class CodeReviewComment < ApplicationRecord
 
   before_save :compute_is_from_teacher
 
-  # To do: move to ReviewableProject model
-  def self.user_can_review_project?(project_owner, potential_reviewer)
+  def self.user_can_review_project?(project_owner, potential_reviewer, storage_app_id, level_id = nil, script_id = nil)
+    # user can always review own project
     return true if project_owner == potential_reviewer
+    # teacher can always review student projects
     return true if project_owner.student_of?(potential_reviewer)
+    # peers can only review projects where code review has been enabled, which creates a ReviewableProject
+    reviewable_projects = ReviewableProject.where(storage_app_id: storage_app_id, user_id: project_owner.id)
+    if level_id
+      reviewable_projects = reviewable_projects.where(level_id: level_id)
+    end
+    if script_id
+      reviewable_projects = reviewable_projects.where(script_id: script_id)
+    end
```

Comment:
for my understanding, why do we need to check for these separately? is it just because one of them may be nil?

Reply:
it's not guaranteed we will have both so I split them out.